src/Admin/AdminHelper.php

```diff
@@ -231,6 +229,19 @@ public function addNewInstance($object, FieldDescriptionInterface $fieldDescript
     {
         $instance = $fieldDescription->getAssociationAdmin()->getNewInstance();
         $mapping = $fieldDescription->getAssociationMapping();
+        $parentMappings = $fieldDescription->getParentAssociationMappings();
+
+        foreach ($parentMappings as $parentMapping) {
+            $method = sprintf('get%s', Inflector::classify($parentMapping['fieldName']));
+
+            if (!method_exists($object, $method)) {
+                throw new \RuntimeException(
+                    sprintf('Please add a method %s in the %s class!', $method, ClassUtils::getClass($object))
```

Comment:
I know there is a previous exception like this using a different type and different wording, but IMO this way is more accurate:
```suggestion
            if (!is_callable([$object, $method])) {
                throw new \BadMethodCallException(
                    sprintf('Method %s::%s() does not exist.', ClassUtils::getClass($object), $method)
```

Reply:
Should I make the changes for the other exception too @phansys ?

Reply:
I think yes, regardless technically it will be a BC break (`\BadMethodCallException` does not extend `\RuntimeException`).
Let's wait for more opinions.